backend/mlarchive/bin/compare.py

```diff
@@ -21,53 +21,58 @@
 from mlarchive.bin.scan_utils import all_mboxs
 from mlarchive.archive.models import Message
 from mlarchive.archive.mail import archive_message, MessageWrapper
+from zoneinfo import ZoneInfo
 
 TOTAL = 0 
 MISSING = 0
 IMPORTED = 0
 
+pacific = ZoneInfo('US/Pacific')
+
+
 def check_message(message, listname, load):
     global TOTAL, MISSING, IMPORTED
     TOTAL += 1
     
     try:
         msgid = message.get('Message-ID').strip('<>')
-    except AttributeError as e:
+    except AttributeError:
         return
     
     try:
-        x = Message.objects.get(msgid=msgid,email_list__name=listname)
+        _ = Message.objects.get(msgid=msgid, email_list__name=listname)
     except Message.DoesNotExist:
         MISSING += 1
         if load:
             status = archive_message(message.as_string(),
-                                              listname,
-                                              save_failed=False)
+                                     listname,
+                                     save_failed=False)
             if status == 0:
                 status_message = 'imported'
                 IMPORTED += 1
             else:
                 status_message = 'import failed'
         else:
             status_message = ''
-        print("%s, %s, %s, %s" % (listname,message['subject'][:20],message['date'],status_message))
+        print("%s, %s, %s, %s" % (listname, str(message['subject'])[:20], message['date'], status_message))
+
     
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('-s', '--start', 
                         required=True,
-                        help="enter the date to start comparison YYYY-MM-DDTHH:MM:SS")
-    parser.add_argument('-e', '--end', help="enter the date to end comparison YYYY-MM-DDTHH:MM:SS")
+                        help="enter the date to start comparison YYYY-MM-DDTHH:MM (UTC)")
+    parser.add_argument('-e', '--end', help="enter the date to end comparison YYYY-MM-DDTHH:MM (UTC)")
     parser.add_argument('--list', help='restrict comparison to specified list')
     parser.add_argument('-l', '--load', action='store_true', help='import missing messages')
     args = parser.parse_args()
     
-    start_date = parse(args.start)
+    start_date = parse(args.start).astimezone(datetime.timezone.utc)
```

Comment:
In my experience this method is robust. parse returns a naive datetime when not provided with a timezone, which is assigned the default, UTC. When provided with a timezone, it is respected and converted to UTC.
I am not sure why this differs from above.
```
>>> parse("2023-09-22T11:31")
datetime.datetime(2023, 9, 22, 11, 31)
>>> str(parse("2023-09-22T11:31").astimezone(datetime.timezone.utc))
'2023-09-22 11:31:00+00:00'
>>> str(parse("2023-09-22T11:31Z").astimezone(datetime.timezone.utc))
'2023-09-22 11:31:00+00:00'
```
Which works well when supported timezone syntax is used.
```
>>> str(parse("2023-09-22T11:31-07:00").astimezone(datetime.timezone.utc))
'2023-09-22 18:31:00+00:00'
```
I can switch to use fromisoformat if that helps.

Reply:
I believe the difference is because the default that `parse()` uses is the local timezone of the machine it's running on. On my system, it's ADT and I get
```python
>>> str(parse("2023-09-22T11:31").astimezone(datetime.timezone.utc))  # system time is ADT
'2023-09-22 14:31:00+00:00'
```
on ietfa, it's PDT and I get
```python
>>> str(parse("2023-09-22T11:31").astimezone(datetime.timezone.utc))  # system time is PDT
'2023-09-22 18:31:00+00:00'
```
and on your system, time is apparently UTC and the result matches. I don't think there's any way to tell it what to use as the default, it just uses system time.

If you want to guarantee that it's either UTC or whatever the user specified, you probably have to do an explicit check for whether you got a naive datetime and `replace()` only if it doesn't already have a tzinfo. Most succinct I can think of is
```python
dt = parse(...)
dt = dt.replace(tzinfo=datetime.timezone.utc if dt.tzinfo is None else dt.tzinfo)
```

`fromisoformat()` has the same behavior with respect to system time, it just saves you a dependency if you only mean to allow the iso format and don't want the additional flexibility that `dateutil` affords.

Reply:
(docker containers typically run in UTC regardless of what system they're on, I believe)